src/pkg/packager/composer/list.go

```diff
@@ -72,14 +103,14 @@ func (ic *ImportChain) append(c types.ZarfComponent, relativeToHead string, vars
 }
 
 // NewImportChain creates a new import chain from a component
-func NewImportChain(head types.ZarfComponent, arch, flavor string) (*ImportChain, error) {
+func NewImportChain(head types.ZarfComponent, index int, arch, flavor string) (*ImportChain, error) {
```

Comment:
Anywhere we return an error in this function, we should also return a `nil` *ImportChain pointer

Lines:

- 130
- 135
- 139
- 153
- 160
- 165
- 172
- 189
- 191
- 195
- 197